spacy_llm/tasks/srl_task.py

```diff
@@ -0,0 +1,407 @@
+
+    pred_label2relations = _get_label2rels(pred_relation_tuples)
+    gold_label2relations = _get_label2rels(gold_relation_tuples)
+
+    all_labels = set.union(
+        set(pred_label2relations.keys()), set(gold_label2relations.keys())
+    )
+    label2prf = {}
+    for label in all_labels:
+        pred_label_rels = pred_label2relations[label]
+        gold_label_rels = gold_label2relations[label]
+        label2prf[label] = _overlap_prf(gold_label_rels, pred_label_rels)
+
+    return {
+        "Predicates": predicates_prf,
+        "ARGs": {"Overall": micro_rel_prf, "PerLabel": label2prf},
+    }
+
+
+@registry.llm_tasks("spacy.SRL.v1")
+def make_srl_task(
+    labels: Union[List[str], str] = [],
+    template: str = _DEFAULT_SPAN_SRL_TEMPLATE_V1,
+    label_definitions: Optional[Dict[str, str]] = None,
+    examples: ExamplesConfigType = None,
+    normalizer: Optional[Callable[[str], str]] = None,
+    alignment_mode: Literal["strict", "contract", "expand"] = "contract",
+    case_sensitive_matching: bool = True,
+    single_match: bool = True,
+    verbose: bool = False,
+    predicate_key: str = "Predicate",
+):
+    """SRL.v1 task factory.
+
+    labels (str): Comma-separated list of labels to pass to the template.
+        Leave empty to populate it at initialization time (only if examples are provided).
+    template (str): Prompt template passed to the model.
+    label_definitions (Optional[Dict[str, str]]): Map of label -> description
+        of the label to help the language model output the entities wanted.
+        It is usually easier to provide these definitions rather than
+        full examples, although both can be provided.
+    examples (Optional[Callable[[], Iterable[Any]]]): Optional callable that
+        reads a file containing task examples for few-shot learning. If None is
+        passed, then zero-shot learning will be used.
+    normalizer (Optional[Callable[[str], str]]): optional normalizer function.
+    alignment_mode (str): "strict", "contract" or "expand".
+    case_sensitive_matching: Whether to search without case sensitivity.
+    single_match (bool): If False, allow one substring to match multiple times in
+        the text. If True, returns the first hit.
+    verbose (boole): Verbose ot not
+    predicate_key: The str of Predicate in the template
+    """
+    labels_list = split_labels(labels)
+    raw_examples = examples() if callable(examples) else examples
+    rel_examples = [SRLExample(**eg) for eg in raw_examples] if raw_examples else None
+    return SRLTask(
+        labels=labels_list,
+        template=template,
+        label_definitions=label_definitions,
+        examples=rel_examples,
+        normalizer=normalizer,
+        verbose=verbose,
+        alignment_mode=alignment_mode,
+        case_sensitive_matching=case_sensitive_matching,
+        single_match=single_match,
+        predicate_key=predicate_key,
+    )
+
+
+class SRLTask(SerializableTask[SRLExample]):
```

Comment:
Like `class SRLTask(SpanTask[SRLExample]):` ?

Reply:
https://github.com/explosion/spacy-llm/pull/226/commits/cb81bdf53f4e020e256af98c2a4be83d8bb2dbaa

Reply:
Yes, like `class SRLTask(SpanTask[SRLExample])`. https://github.com/explosion/spacy-llm/commit/cb81bdf53f4e020e256af98c2a4be83d8bb2dbaa has `SpanTask` inherit from `Generic[ExampleType]` though - why do you think that's necessary?

Reply:
I think I was getting an error without it. Maybe it is required to allow inheritance?

`E   TypeError: <class 'spacy_llm.tasks.span.SpanTask'> is not a generic class`

Reply:
https://github.com/explosion/spacy-llm/blob/ca19b3f72bf5ab9e8a6aadaedfbd40621057733a/spacy_llm/tasks/span.py#L48

This is breaking inheritance For SRLTask

Reply:
How so? Don't we have the same label consistency issue with `SRLTask` as well?

Reply:
nevermind, needed to implement a `self._check_label_consistency` for `SRLTask`. It was otherwise running the Span's routine, which ran on `SpanExample` instead of `SRLExample`

Reply:
https://github.com/explosion/spacy-llm/pull/226/commits/8fc6b8d44c106a794ec0b2743b6b787e1683b37f